src/picosvg/svg.py

```diff
@@ -293,18 +303,32 @@ def _resolve_use(self, scope_el):
                 ref = use_el.attrib.get(_xlink_href_attr_name(), "")
                 if not ref.startswith("#"):
                     raise ValueError(f"Only use #fragment supported, reject {ref}")
-                target = self.xpath_one(f'//svg:*[@id="{ref[1:]}"]')
+
+                target = el_by_id.get(ref[1:], None)
+                if target is None:
+                    raise ValueError(f"No element has id '{ref[1:]}'")
 
                 new_el = copy.deepcopy(target)
+                # leaving id's on <use> instantiated content is a path to duplicate ids
+                for el in new_el.getiterator("*"):
+                    if "id" in el.attrib:
+                        del el.attrib["id"]
 
                 group = etree.Element(f"{{{svgns()}}}g", nsmap=self.svg_root.nsmap)
-                use_x = use_el.attrib.get("x", 0)
-                use_y = use_el.attrib.get("y", 0)
-                if use_x != 0 or use_y != 0:
-                    group.attrib["transform"] = (
-                        group.attrib.get("transform", "")
-                        + f" translate({use_x}, {use_y})"
-                    ).strip()
+                affine = Affine2D.identity().translate(
+                    float(use_el.attrib.get("x", 0)), float(use_el.attrib.get("y", 0))
+                )
+
+                if "transform" in use_el.attrib:
+                    affine = Affine2D.compose_ltr(
+                        (
+                            affine,
+                            parse_svg_transform(use_el.attrib["transform"]),
```

Comment:
can't we just use Affine2D.fromstring and avoid importing parse_svg_transform?

```suggestion
                            Affine2D.fromstring(use_el.attrib["transform"]),
```

Reply:
so IIUC the intention is that the x/y offsets are applied first (if any), and then the transform is applied? 

in the svg spec I read https://svgwg.org/svg2-draft/single-page.html#struct-UseLayout

> The x and y properties define an additional transformation (translate(x,y), where x and y represent the computed value of the corresponding property) to be applied to the ‘use’ element, _after_ any transformations specified with other properties (i.e., _appended to the right-side_ of the transformation list).

I think since our method composes LTR they way you're doing is correct, but wanted to double check with you

Reply:
That is my understanding as well. Also I tested doing it the other way around and confirmed the stars go in the wrong places :)